src/fs/driver/fat/fat_common.c

```diff
@@ -397,11 +413,10 @@ uint32_t fat_get_fat_(struct fat_fs_info *fsi,
 			result = (uint32_t) p_scratch[offset] |
 			  ((uint32_t) p_scratch[offset+1]) << 8;
 		}
-		if (cluster & 1) {
+		if (cluster & 1)
```

Comment:
Probably, that's related to fact, that for FAT12 size of cluster entry (12 bits) is not multiple of byte size (8 bits), so three bytes store info about 2 clusters (12 bit for each cluster, 8 bit per byte), and align is required.